String.php

```diff
@@ -767,6 +767,68 @@ public static function getCharDirection ( $char ) {
     }
 
     /**
+     * Get the number of column positions of a wide-character.
+     *
+     * This is a PHP implementation of wcwidth() and wcswidth() (defined in IEEE
+     * Std 1002.1-2001) for Unicode, by Markus Kuhn. Please, see
+     * http://www.cl.cam.ac.uk/~mgk25/ucs/wcwidth.c.
+     *
+     * The wcwidth(wc) function shall either return 0 (if wc is a null
+     * wide-character code), or return the number of column positions to be
+     * occupied by the wide-character code wc, or return -1 (if wc does not
+     * correspond to a printable wide-character code).
+     *
+     * @access  public
+     * @param   string  $char    Character.
+     * @return  int
+     */
+    public static function getCharWidth ( $char ) {
+
+        $char = (string) $char;
+        $c    = static::toCode($char);
+
+        // Test for 8-bit control characters.
+        if(0x0 === $c)
+            return 0;
+
+        if(0x20 > $c || (0x7f <= $c && $c < 0xa0))
+            return -1;
+
+        // Non-spacing characters.
+        if(   0xad !== $c
+           && 0    !== preg_match('#^[\p{Mn}\p{Me}\p{Cf}\x{1160}-\x{11ff}\x{200b}]$#u', $char))
```

Comment:
imho you still need to remove the $ anchor. One anchor (^) is enough to get the speed benefit, but moreover it will make your method's behavior consistent with length>1 strings (because nothing prevents anyone from doing so, you should define a behavior you want and add tests for this case also).